src/sized_chunk/mod.rs

```diff
@@ -583,37 +589,42 @@ where
         if self.right == N::USIZE || (self.left >= insert_size && left_size < right_size) {
             unsafe {
                 Chunk::force_copy(self.left, self.left - insert_size, left_size, self);
-                let mut write_index = real_index - insert_size;
+                let write_index_start = real_index - insert_size;
+                let mut written = 0;
                 for value in iter {
-                    Chunk::force_write(write_index, value, self);
-                    write_index += 1;
+                    Chunk::force_write(write_index_start + written, value, self);
+                    written += 1;
                 }
+                // update after writes for panic safety
+                // update actual number written in case iterator stopped earlier than promised
+                self.left -= written;
```

Comment:
Iterator that terminates early can still cause memory safety problem.

Consider the scenario where `insert_from(0)` is called with an iterator that reports size 3 but only returns one element 5 and terminates, with the chunk state like this: [x, x, x, (1, 2, 3)], parentheses mean left and write.

1. It falls into this first branch - insert_size=3, index=0, real_index=3, left_size=0, right_size=3, write_index_start=0
2. `force_copy()` duplicates elements [1, 2, 3, (1, 2, 3)]
3. First iteration [5, 2, 3, (1, 2, 3)]
4. Iteration finishes, self.left -= 1: [5, 2, (3, 1, 2, 3)]

Element 3 will cause double-drop